### PR TITLE
Option to use files or server for probes. Run integration tests using Makefile.

### PR DESCRIPTION
**Pull request by nikos912000**
### :pencil: Description

- Introduces flags to enable/disable file and server probes. Makes writing to files the default option.
- Adds option to run integration tests using the Makefile.
- Updates docs.

### :link: Related Issues
Implements #11.

### CODE REVIEW

File: docs/about/getting-started.md [hunk 1]
```diff
@@ -5,37 +5,39 @@ title: Getting Started
...
+| --grpc-requests                    | strings | N/A              | gRPC request to be sent. Request is in '\<service\>\<method\>\[:message\]' format. E.g. health/ping:{"key": "value"}    |
+| --help, -h                         |         | N/A              | Help for warmup-sidecar                                                                                                  |
+| --http-headers                     | strings | N/A              | Http headers to be sent with warm up requests.                                                                           |
+| --http-requests                    | strings | N/A              | Http request to be sent. Request is in '\<http-method\>:\<path\>\[:body\]' format. E.g. post:/ping:{"key": "value"}      |
+| --probe-file-enabled               | bool    | true             | If set to true writes files to be used as readiness/liveness probes                                                       |
+| --probe-file-liveness-path         | string  | alive            | File to be used for liveness probe                                                                                       |
```
Comment (worldtiki) [inline docs/about/getting-started.md]: It doesn't seem necessary to allow users to specify the names of these files. We can just document their values (ready and alive).

File: Makefile [hunk 2]
```diff
@@ -1,14 +1,17 @@
-.PHONY: all build test
+.PHONY: all build unit-tests integration-tests
```
Comment (worldtiki) [inline Makefile]: There's not a lot of tests, can we keep just build + tests?

Comment (nikos912000) [reply Makefile]: The main reason for separating those was that users expect the project to build fast. As part of the integration tests we spin up a target server which is time-consuming.

Also, planning to add more end-to-end tests and validation around those so better to do this now.

File: docs/about/getting-started.md (hunk 1 above)
Comment (nikos912000) [reply docs/about/getting-started.md]: It's true that it seems unnecessary. 
However, whenever it comes to files written in disk from a third-party library (like Mittens) I think users should have the option to decide on the name/filepath. This also gives them the option to write on a particular directory which would be handy from a permissions perspective.

Comment (worldtiki) [reply docs/about/getting-started.md]: That's true. I was thinking of the use case where this is used as a docker sidecar (and the file is written inside the container).

**Pull request merged**